src/abstractdataframe/selection.jl

```diff
@@ -0,0 +1,361 @@
+    copy!(_names(x), _names(newdf))
+    empty!(x.lookup)
+    for (i, n) in enumerate(x.names)
+        x.lookup[n] = i
+    end
+    return df
+end
+
+"""
+    select(df::AbstractDataFrame, inds...; copycols::Bool=true)
+
+Create a new data frame that contains columns from `df`
+specified by `inds` and return it.
+
+Arguments passed as `inds...` can be any index that is allowed for column indexing
+provided that the columns requested in each of them are unique and present in `df`.
+In particular, regular expressions, `All`, `Between`, and `Not` selectors  are supported.
+
+Also if `df` is a `DataFrame` or `copycols=true` then column renaming and transformations
+are supported. The syntax for column renaming is `old_column=>new_column_name`.
+The syntax for column transformations is `old_column=>fun=>new_column_name`.
+If `old_column` is a `Symbol` or an integer then and `fun` must be callable
+that is applied row by row to the values of `old_column`.
+Otherwise `old_column` can be any column indexing syntax, but in this case `fun`
+will be passed `DataFrameRows` selected by `old_column`.
+
+Column transfomation syntax also supports a short `old_column=>fun` form, in which
+case `new_column_name` is automatically generated by joining `olcd_column` with `fun` name
+with `_`.
+
+If more than one argument is passed then duplicates are accepted except for
+column renaming and transformation operations, where it is not allowed to rename/transform
+into the same column name.
+For example if `:col` is present in `df` a call `select(df, :col, :)` is valid
+and creates a new data frame with column `:col` moved to be the first.
+
+If `df` is a `DataFrame` a new `DataFrame` is returned.
+If `copycols=true` (the default), then returned `DataFrame` is guaranteed not to share columns with `df`.
+If `copycols=false`, then returned `DataFrame` shares column vectors with `df` where possible.
+
+If `df` is a `SubDataFrame` then a `SubDataFrame` is returned if `copycols=false`
+and a `DataFrame` with freshly allocated columns otherwise.
+
+Note that if `df` is a `DataFrame` and `copycols=false` then including the same column several times
+in the resulting data frame will create aliases.
+
+# Examples
+```jldoctest
+julia> df = DataFrame(a=1:3, b=4:6)
+3×2 DataFrame
+│ Row │ a     │ b     │
+│     │ Int64 │ Int64 │
+├─────┼───────┼───────┤
+│ 1   │ 1     │ 4     │
+│ 2   │ 2     │ 5     │
+│ 3   │ 3     │ 6     │
+
+julia> select(df, :b)
+3×1 DataFrame
+│ Row │ b     │
+│     │ Int64 │
+├─────┼───────┤
+│ 1   │ 4     │
+│ 2   │ 5     │
+│ 3   │ 6     │
+
+julia> select(df, Not(:b)) # drop column :b from df
+3×1 DataFrame
+│ Row │ a     │
+│     │ Int64 │
+├─────┼───────┤
+│ 1   │ 1     │
+│ 2   │ 2     │
+│ 3   │ 3     │
+
+julia> select(df, :a=>:c, :b)
+3×2 DataFrame
+│ Row │ c     │ b     │
+│     │ Int64 │ Int64 │
+├─────┼───────┼───────┤
+│ 1   │ 1     │ 4     │
+│ 2   │ 2     │ 5     │
+│ 3   │ 3     │ 6     │
+
+julia> select(df, :b, :a=><(1.5)=>:c)
```

Comment:
This might be a bit too clever for newcomers who don't know the  `<(x)` syntax. Maybe use something more standard like `abs` or `sin`? It would be also be useful to show a custom anonymous function with several arguments.

Also I find it easier to read with spaces around `=>` (especially if we add `x -> ...` to the picture). What do you think?

Reply:
ok. fixed